runtime/permissions.rs

```diff
@@ -1126,6 +1126,156 @@ pub struct PermissionsOptions {
   pub prompt: bool,
 }
 
+impl<'de> Deserialize<'de> for PermissionsOptions {
+  fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+  where
+    D: Deserializer<'de>,
+  {
+    struct UnaryPermissionsOptions(Option<Vec<String>>);
+    impl<'de> Deserialize<'de> for UnaryPermissionsOptions {
+      fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+      where
+        D: Deserializer<'de>,
+      {
+        struct UnaryPermissionsOptionsVisitor;
+        impl<'de> de::Visitor<'de> for UnaryPermissionsOptionsVisitor {
+          type Value = Option<Vec<String>>;
+
+          fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("boolean or string[]")
+          }
+
+          fn visit_bool<E>(self, v: bool) -> Result<Self::Value, E>
+          where
+            E: de::Error,
+          {
+            if v {
+              Ok(Some(vec![]))
+            } else {
+              Ok(None)
+            }
+          }
+
+          fn visit_seq<V>(self, mut v: V) -> Result<Self::Value, V::Error>
+          where
+            V: de::SeqAccess<'de>,
+          {
+            let mut granted_list = vec![];
+            while let Some(value) = v.next_element::<String>()? {
+              granted_list.push(value);
+            }
+            Ok(Some(granted_list))
+          }
```

Comment:
should encountering an empty seq return `None`? because ie
```json
"permissions": {
    "net": [],
  }
  ```
  is the same as `"net": true`, though an empty array seems more like `"net": false` behaviour would be expected